### PR TITLE
fix(DataGrid): fixed overflowing cell content 

### PR DESCRIPTION
**Pull request by ChristianBusshoff**
Relates to #5900

### CODE REVIEW

File: packages/sit-onyx/src/components/OnyxCalendar/OnyxCalendar.vue [hunk 1]
```diff
@@ -404,6 +403,7 @@ useOutsideClick({
         background-color: var(--onyx-color-base-background-blank);
 
         th {
+          overflow-wrap: normal;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxCalendar/OnyxCalendar.vue]: Nice that fixes our current issue in the showcase where the weekdays are cut off :)

Image

But I am not sure if this is the correct fix since the issue still seems to be that the weekday cells do not have enough width for their content, thats why they overflow at all. Would be great to adjust the width instead

Image

File: packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue [hunk 2]
```diff
@@ -242,9 +242,9 @@ const headlineId = computed(() => (slots.headline ? _headlineId : undefined));
         &:hover:not(.onyx-table__empty, .onyx-table__standalone-row) > td::before {
           background-color: var(--onyx-color-base-neutral-200);
         }
-
         > th,
         > td {
+          overflow-wrap: break-word;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxTable/OnyxTable.vue:247]: Could you please add a screenshot test for the cell and header overflow? So that we can ensure they behave correctly

File: .changeset/stale-hands-attend.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": major
+---
+
+fix(OnyxTable/OnyxDataGrid): Prevent long words from overflowing table cells.
```
Comment (larsrickert) [inline .changeset/stale-hands-attend.md]:
```suggestion
- fix(OnyxTable/OnyxDataGrid): Prevent long words from overflowing table cells.
- fix(OnyxCalendar): ensure weekday labels do not wrap

```

File: .changeset/stale-hands-attend.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"sit-onyx": major
```
Comment (larsrickert) [inline .changeset/stale-hands-attend.md]:
```suggestion
"sit-onyx": patch
```

We do not want to publish onyx 2.0 with this 😅

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png
Comment (larsrickert) [inline packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png:1]: Is this change intended? Shouldn't the header actions be placed directly next to the label?

File: packages/sit-onyx/src/components/OnyxCalendar/OnyxCalendar.vue [hunk 5]
```diff
@@ -371,6 +371,8 @@ useOutsideClick({
     $calendar-week-column-width: 2.5rem;
     $calendar-day-number-display-width: 2rem;
 
+    min-width: 18rem;
```
Comment (larsrickert) [inline packages/sit-onyx/src/components/OnyxCalendar/OnyxCalendar.vue:374]: The min width might not always work, e.g. when switching the other languages where the weekday names are different. We can keep this for now tho but ideally the min-width is based on the content

**Review by larsrickert — CHANGES_REQUESTED** [review]

File: packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png
Comment (ChristianBusshoff) [reply packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png:1]: If the label doesn't get a width of 100% the overflowing is not working. That's why i changed it.

Comment (larsrickert) [reply packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png:1]: But then we should make sure that the content is still placed next to the label

Comment (ChristianBusshoff) [reply packages/sit-onyx/playwright/snapshots/components/OnyxDataGrid/features/base/headline-filtered-chrome-linux.png:1]: Okay, by setting max-width instead it's also working. I change it.

**Review by larsrickert — APPROVED** [review]
